### ISSUE TITLE
Keeps asking for a force subtitles parameter in the settings

### ISSUE THREAD

**Issue by vampjoseph** [jaw20/Crunchyroll-XML-Decoder#10]
At this point after months of trying to configure python to work with the base toolkit and your update to make it work with the site again, I seem to have hit yet another wall that I can't figure out.

I've been testing the current config of python that I've been able to cobble together (I know pretty well nothing about coding much less coding in python and have been getting by on manuals and forum threads) and I've gotten to the point where it's whining about the downloaders settings file which is great news for me as it means I'm heading in the right direction. The first problem was an easy fix, it wanted a second language setting so I added one to the .ini. Now it wants a force subtitles setting which I've added but it wants parameters which I don't know or don't have and it won't accept a boolean search (i.e.: forcesubtitle = English)

This is the error message without the setting:

Traceback (most recent call last):
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", line 376, in <module>
    ultimate(page_url, seasonnum, epnum)
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", line 221, in ultimate
    lang1, lang2, forcesub, forceusa, localizecookies, vquality, onlymainsub = altfuncs.config()
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\altfuncs.py", line 32, in config
    forcesub = configr.getboolean('SETTINGS', 'forcesubtitle')
  File "C:\Python27\lib\ConfigParser.py", line 368, in getboolean
    v = self.get(section, option)
  File "C:\Python27\lib\ConfigParser.py", line 618, in get
    raise NoOptionError(option, section)
ConfigParser.NoOptionError: No option 'forcesubtitle' in section: 'SETTINGS'
An error has occured. Please check "error.log" for the affected page url.

This is with the setting added but without a parameter:

Traceback (most recent call last):
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", line 376, in <module>
    ultimate(page_url, seasonnum, epnum)
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", line 221, in ultimate
    lang1, lang2, forcesub, forceusa, localizecookies, vquality, onlymainsub = altfuncs.config()
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\altfuncs.py", line 17, in config
    configr.read('settings.ini')
  File "C:\Python27\lib\ConfigParser.py", line 305, in read
    self._read(fp, filename)
  File "C:\Python27\lib\ConfigParser.py", line 546, in _read
    raise e
ConfigParser.ParsingError: File contains parsing errors: settings.ini
        [line 13]: 'forcesubtitle'
An error has occured. Please check "error.log" for the affected page url.

Thanks for reading down to this point and I apologize for hitting you with a wall of text. I look forward to a response

**Comment by jaw20** [comment]
> At this point after months of trying to configure python

I would try and reinstall a clean copy of python first. Make sure that you uninstall the current python version you have and install 2.7.9. Then follow the instructions on the readme. If you still get the error we can go from there.

**Comment by vampjoseph** [comment]
ok did a fresh install of python 2.7.9 with the proper versions of lxml, etc. Now it's bugging me for forceusa parameters. Here is the error message:

Traceback (most recent call last):
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", l
ine 376, in <module>
    ultimate(page_url, seasonnum, epnum)
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\ultimate.py", l
ine 213, in ultimate
    page_url = altfuncs.vidurl(page_url, False, False)
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\altfuncs.py", l
ine 119, in vidurl
    res = gethtml(url)
  File "F:\Crunchyroll Downloader Toolkit DX\crunchy-xml-decoder\altfuncs.py", l
ine 64, in gethtml
    if not forceusa and localizecookies:
NameError: global name 'forceusa' is not defined
An error has occured. Please check "error.log" for the affected page url.

**Comment by jaw20** [comment]
Can you please post the error.log file for me?

**Comment by vampjoseph** [comment]
I am using the latest version as of last month. I have been all over the filetree of the folder including subfolders and haven't found anything that even remotely looks like an error.log file for the default downloader. Either it's saved somewhere else that I haven't found, or it doesn't get generated at all (I even did a file search and the only error.log that I found was for another program). 

I ended up looking at the .py for something to do (as I've tried everything else and was pleasantly surprised to find it was written in plain english) to try and find out what those settings are looking for and found instructions for them between lines 40-65. I guessed what those settings mean (forcesubtitle = true forceusa = false localizecookies = yes onlymainsub = true) and ran a test to see if I would actually be able to download something or just get an error. The test was a success. I ended up with a useable .mkv in the quality I wanted with english subtitles muxed in and properly labeled with the show name, episode number, and the title as a bonus (I wasn't expecting that last bit). 

So my suggestion is if you could either include a settings file or explain the settings in the readme to avoid frustration. I'm attaching my current settings file for your consideration. Thanks for your help so far.

Best Regards
VampJoseph
[settings.zip](https://github.com/jaw20/Crunchyroll-XML-Decoder/files/752491/settings.zip)

**Comment by jono123** [comment]
There is an error on syntax to check cfscrape are installed or not.
I delete that function and manual install the cfscrape package.

But login failed. It seem cannot bypass the cloudfare.

**Comment by jaw20** [comment]
@jono123
 Please make a new ticket, as this isn't related to this issue.
